types/dragino/lsn50v2-d22/uplink.js

```diff
@@ -0,0 +1,156 @@
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const bytes = Hex.hexToBytes(payload);
+
+  const mode = (bytes[6] & 0x7c) >> 2;
+
+  let topic = "default";
+  const data = {};
+  const lifecycle = {};
+
+  if (mode !== 2 && mode !== 31) {
+    lifecycle.batteryVoltage = ((bytes[0] << 8) | bytes[1]) / 1000;
+    let batteryLevel =
+      Math.round((lifecycle.batteryVoltage - 2.45) / 0.0115 / 10) * 10;
+
+    if (batteryLevel > 100) {
+      batteryLevel = 100;
+    } else if (batteryLevel < 0) {
+      batteryLevel = 0;
+    }
+    lifecycle.batteryLevel = batteryLevel;
+
+    data.temperature = parseFloat(
+      ((((bytes[2] << 24) >> 16) | bytes[3]) / 10).toFixed(2),
+    );
+
+    data.c0adc = ((bytes[4] << 8) | bytes[5]) / 1000;
+
+    data.digitalStatus = bytes[6] & 0x02 ? "HIGH" : "LOW";
+
+    if (mode !== 6) {
+      data.extTrigger = !!(bytes[6] & 0x01);
+      data.open = !!(bytes[6] & 0x80);
+    }
+  }
+
+  if (mode === "0") {
```

Comment:
mode does not seem to be string, please recheck the full PR for these occurrences, it seems there are several instances with type mismatches